service/storage_service.hh

```diff
@@ -881,6 +881,7 @@ public:
     // Waits for a topology request with a given ID to complete and return non empty error string
     // if request completes with an error
     future<sstring> wait_for_topology_request_completion(utils::UUID id);
+    future<> wait_for_topology_not_busy();
```

Comment:
nit: patch title says "topo sm"
only you guys really know what that means.
please use descriptive names rather than abbreviations

Reply:
also please add more documentation to the patch log message
it's too terse.
I think it can also be safely squashed with the patch that introduces the rest of the code
I don't see much value in separating it as it probably fixes an issue with the original code (that is introduced in this series anyhow)